Classes/Service/NodeRedirectService.php

```diff
@@ -209,4 +246,22 @@ protected function getUriBuilder()
         }
         return $this->uriBuilder;
     }
+
+    /**
+     * Get the given node in the given dimensions.
+     * If it doesn't exist the method returns null.
+     *
+     * @param NodeInterface $node
+     * @param array $dimensions
+     * @return NodeInterface|null
+     */
+    protected function getNodeInDimensions(NodeInterface $node, array $dimensions)
+    {
+        $context = $this->contextFactory->create([
+            'workspaceName' => $node->getWorkspace()->getName(),
+            'dimensions' => $dimensions,
+            'invisibleContentShown' => true,
+        ]);
+        return $context->getNode($node->getPath());
```

Comment:
I experienced a very strange behaviour here:
using `$context->getNodeByIdentifier($node->getIdentifier())` randomly returned `null` when running the tests multiple times. Using `getNode` however works just fine.